### PR TITLE
Updated intro to runtime detection

### PR DESCRIPTION
**Pull request by slashben**

### CODE REVIEW

File: docs/docs/operator/runtime-threat-detection.md [hunk 1]
```diff
@@ -1,93 +1,120 @@
...
+
+It is based on eBPF-based event monitoring on Kubernetes nodes and an advanced rule engine that allows you to write rules that can detect and respond to threats in your cluster and more specifically in your workload itself🛡️.
+
+## Architecture
+
+Runtime threat detection is part of the Kubescape Operator, and most of the functionality is implemented as part of the Node-agent.
+
+![Node agent design](/assets/node-agent-design.svg)
+
+Node-agent uses [Inspektor Gadget](https://www.inspektor-gadget.io/) for eBPF event aquistion. It uses the [Kubescape Storage](https://github.com/kubescape/storage) to store and monitor detection related objects.
```
Comment (amitschendel) [inline docs/docs/operator/runtime-threat-detection.md:15]: Add that we have also custom eBPF.

**Review by amitschendel — APPROVED** [review]

File: docs/docs/operator/runtime-threat-detection.md [hunk 2]
```diff
@@ -1,93 +1,120 @@
...
+
+### Behavior analysis engine
+Additionally, Kubescape is equipped with rules designed to identify well-known attack signatures. These rules are adept at uncovering various threats, such as unauthorized software executions that deviate from the original container image, detection of unpackers in memory, reverse shell activities, and more. Users have the flexibility to create 'Rule Bindings'—specific instructions that direct Kubescape on which rules should be applied to which Pods. This level of customization ensures that security measures are tailored to the unique needs of each Kubernetes deployment, enhancing the overall security posture and responsiveness of the system.
+
+### Malware scanning
+Kubescape can scan the nodes for malware using ClamAV as an engine, a popular open-source antivirus engine. ClamAV supports scanning of files, directories, and volumes, and can be configured to scan the entire node or only specific directories. You can read more about ClamAV here.
+
+Kubescape uses its own virus database which is a subset of the latest ClamAV virus database release but adopted to Kubernetes environment to save resources.
```
Comment (Oshratn) [inline docs/docs/operator/runtime-threat-detection.md:43]:
```suggestion
Kubescape uses its own virus database which is a subset of the latest ClamAV virus database release but adapted to Kubernetes environment to save resources.
```

File: docs/docs/operator/runtime-threat-detection.md [hunk 3]
```diff
@@ -1,93 +1,120 @@
...
+helm upgrade --install kubescape kubescape/kubescape-operator -n kubescape --create-namespace --set capabilities.runtimeDetection=enable --set alertCRD.installDefault=true
 ```
-To customize these values, you can set them when installing the chart:
+
+This installs the Node-agent with detection enabled and with default alert rule bindings.
+Note, that we have not defined an exporter, therefore the alerts are sent to the Node-agent logs.
+
+Wait for the learning period to end (the default is 24h!) and you can try out the detection capabilities.
+
+For example, in this case we run the system on a Kind cluster and here is what you see in logs if you open a process on a container:
```
Comment (Oshratn) [inline docs/docs/operator/runtime-threat-detection.md:68]:
```suggestion
For example, in this case we ran the system on a Kind cluster and here is what you see in logs if you open a process on a container:
```

File: docs/docs/operator/runtime-threat-detection.md [hunk 4]
```diff
@@ -1,93 +1,120 @@
...
---set nodeAgent.config.updatePeriod=Xm
+$ kubectl apply -f https://k8s.io/examples/application/deployment.yaml
+$ sleep $LEARNING_PERIOD
+$ kubectl exec nginx-deployment-86dcfdf4c6-lh6cb -- echo "I am alerting"
+I am alerting
 ```
 
-### Behavior analysis engine
-The behavior analysis engine is responsible for analyzing the behavior of the application and determining if it is malicious. It does this by analyzing the behavior of the application and comparing it to known malicious behavior. If any malicious behavior is detected, the engine will raise an alert.
+We can see the alerts that were produced due to the fact that the `echo` process is not expect to run in this container:
```
Comment (Oshratn) [inline docs/docs/operator/runtime-threat-detection.md:76]:
```suggestion
We can see the alerts that were produced due to the fact that the `echo` process is not expected to run in this container:
```

File: docs/docs/operator/runtime-threat-detection.md [hunk 5]
```diff
@@ -1,93 +1,120 @@
...
+
+## Architecture
+
+Runtime threat detection is part of the Kubescape Operator, and most of the functionality is implemented as part of the Node-agent.
+
+![Node agent design](/assets/node-agent-design.svg)
+
+Node-agent uses [Inspektor Gadget](https://www.inspektor-gadget.io/) for eBPF event aquistion. It uses the [Kubescape Storage](https://github.com/kubescape/storage) to store and monitor detection related objects.
+
+Alerts can be sent to multiple sink components, from logs to Prometheus AlertManager.
```
[the remark this answers is not in the document]
Comment (slashben) [reply docs/docs/operator/runtime-threat-detection.md:17]: no, sink as a data engineering concept

**Review by idohuber — APPROVED** [review]
gezer

**Pull request merged**